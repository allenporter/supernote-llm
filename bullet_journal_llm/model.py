"""Data model for bullet journal llm."""

import pathlib

import yaml
from dataclasses import dataclass, field
from mashumaro import DataClassDictMixin
from mashumaro.config import BaseConfig

@dataclass
class Prompt(DataClassDictMixin):
    """A bullet journal prompt."""

    prompt: str | None = None
    filename: str | None = None
    content: str | None = None

    class Config(BaseConfig):
        omit_none = False

@dataclass
class RapidLogEntry:
    """A rapid log entry."""

    type: str
    content: str
    status: str | None = None
    label: str | None = None
    critical: bool = False
    date: str | None = None
    entries: list[str] = field(default_factory=list)

    class Config(BaseConfig):
        omit_none = False


@dataclass
class JournalPage(DataClassDictMixin):
    """A parsed notebook entry."""

    filename: str
    created_at: str
    label: str
    date: str
    records: list[RapidLogEntry]

    class Config(BaseConfig):
        omit_none = False


@dataclass
class DynamicPrompt:
    """A dynamic prompt."""

    prompt: Prompt
    pages: list[JournalPage] = field(default_factory=list)

    @classmethod
    def from_file(cls, filename: pathlib.Path) -> "DynamicPrompt":
        """Create a dynamic prompt from a file."""
        content = filename.read_text()
        docs = list(yaml.load_all(content, Loader=yaml.CSafeLoader))
        if not docs:
            raise ValueError(f"Failed to parse {filename}")

        prompt = Prompt.from_dict(docs[0])
        pages = [JournalPage.from_dict(doc) for doc in docs[1:]]

        return cls(prompt=prompt, pages=pages)
