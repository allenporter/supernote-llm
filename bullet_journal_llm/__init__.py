"""Package for a Bullet Journal in a Supernote notebook."""
