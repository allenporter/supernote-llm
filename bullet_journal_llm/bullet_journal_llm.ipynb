{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Bullet Journal LLM\n",
    "\n",
    "I currently use the SuperNote as a [Bullet Journal](https://www.youtube.com/watch?v=fm15cmYU0IM). This\n",
    "notebook is experimenting with processing the supernote media, using extracted text from note pages,\n",
    "and determining what we can infer or cleanup about each note."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Simple Daily Analysis\n",
    "\n",
    "Look at an arbitrary day's note and see what can be inferred/cleaned up."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO:__main__:Initializing client library\n"
     ]
    }
   ],
   "source": [
    "from pathlib import Path\n",
    "import logging\n",
    "import sys\n",
    "import yaml\n",
    "\n",
    "import google.generativeai as genai\n",
    "import os\n",
    "\n",
    "_LOGGER = logging.getLogger(__name__)\n",
    "logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)\n",
    "\n",
    "NOTES_DIR = Path(\"../texts\")\n",
    "PAGES_DIR = Path(\"../pages\")\n",
    "SECRETS = Path(\"../secrets.yaml\")\n",
    "\n",
    "MODEL_ID = 'gemini-1.5-flash'\n",
    "\n",
    "secrets = yaml.safe_load(SECRETS.read_text())\n",
    "\n",
    "genai.configure(api_key=secrets[\"gemini_api_key\"])\n",
    "\n",
    "_LOGGER.info(\"Initializing client library\")\n",
    "model = genai.GenerativeModel(MODEL_ID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PosixPath('../texts/Daily-16-P20240201073944998430qrrAJQfuUG0J.txt')"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import random\n",
    "\n",
    "daily_notes = list(NOTES_DIR.glob(\"Daily-*.txt\"))\n",
    "\n",
    "random.shuffle(daily_notes)\n",
    "daily_notes[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0201 THU\n",
      "✗ aot rent\n",
      "- ai throughput notes\n",
      "- Bard evcls :/videos?\n",
      "' Ask Photos user study ,' run synthetic data % lab\n",
      "notes for synthetic data\n",
      "* - review Tl Databases slides\n",
      "wrap up DMA rollout plan\n",
      "' summarize s nth data next\n",
      "' monthly reflection\n",
      "• query lens transitions 1 t 0202 FRI * E serving TPU cal\n",
      "Tristan email intro\n",
      "- 15 mins: DMA rollout plan cleanup\n",
      "1- synth date next steps\n",
      "1. gap print return label\n",
      "- At throughput bullet points\n",
      ". review quality report\n",
      "0203 SAT, 0204 SUN\n",
      "- Boo 1-oct flights to LOW\n",
      ". TP stand\n",
      "Fill cavity endar * .\n"
     ]
    }
   ],
   "source": [
    "with daily_notes[0].open() as f:\n",
    "    text = f.read()\n",
    "    print(text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "OCR_PROMPT = \"\"\"You are analyzing OCR text from an e-ink SuperNote A5X. The text has been extract\n",
    "automatically by the device and is not perfect. You can see that the text is not perfect\n",
    "and will need some cleaning up. The text is in English and is in a bullet journal format.\n",
    "The bullet journal can contain a variety of different types of information, such as\n",
    "notes, to-do lists, and reminders. \n",
    "\n",
    "\n",
    "An entry in a daily journal can have different types of bullets:\n",
    "o event\n",
    ". task\n",
    "x completed task\n",
    "- note\n",
    "< moved to future log\n",
    "> moved to another daily, weekly, or monthly note\n",
    "\n",
    "Here are examples of what a daily note looks like:\n",
    "\n",
    "1030 SUN\n",
    "o Halloween Party @ 7PM @ Kid's school \n",
    ". Pick up candy\n",
    "x Buy pumpkins\n",
    "> Call mom\n",
    "\n",
    "1031 MON\n",
    "- Trickortreaters were so cute\n",
    ". Call mom\n",
    "\n",
    "\n",
    "Your task is to rewrite the journal in a consistent format and fix mistakes in\n",
    "the OCR. Only answer with the corrected text, and no other commentary, as the\n",
    "answer will be programtically parsed\n",
    "\n",
    "Original OCR text {filename}:\n",
    "{note}\n",
    "\n",
    "Corrected bullet journal:\n",
    "\"\"\"\n",
    "\n",
    "response = model.generate_content(OCR_PROMPT.format(note=text, filename=str(daily_notes[0])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0201 THU\n",
      "x paid rent\n",
      "- AI throughput notes\n",
      "- Bard evcls :/videos?\n",
      "- Ask Photos user study, run synthetic data % lab\n",
      "notes for synthetic data\n",
      "- review TL Databases slides\n",
      "- wrap up DMA rollout plan\n",
      "- summarize synthetic data next\n",
      "- monthly reflection\n",
      "- query lens transitions 1 to 0202 FRI * E serving TPU calc\n",
      "- Tristan email intro\n",
      "- 15 mins: DMA rollout plan cleanup\n",
      "- synth data next steps\n",
      "- gap print return label\n",
      "- AI throughput bullet points\n",
      "- review quality report\n",
      "0203 SAT, 0204 SUN\n",
      "- Book 1-oct flights to LOW\n",
      "- TP stand\n",
      "- Fill cavity calendar * \n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(response.text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DEBUG:PIL.PngImagePlugin:STREAM b'IHDR' 16 13\n",
      "DEBUG:PIL.PngImagePlugin:STREAM b'IDAT' 41 65536\n"
     ]
    }
   ],
   "source": [
    "import PIL.Image\n",
    "\n",
    "page_filename = PAGES_DIR / (\"_\".join(daily_notes[0].name.split(\"-\")[0:2]) + \".png\")\n",
    "page_png = PIL.Image.open(page_filename)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "IMG_PROMPT = \"\"\"You are analyzing OCR text from an e-ink SuperNote A5X. The note is hand\n",
    "written. You can see that the text is not perfect and will need some cleaning up. The\n",
    "text is in English and is in a bullet journal format. The bullet journal can contain a\n",
    "variety of different types of information, such as notes, to-do lists, and reminders. \n",
    "\n",
    "An entry in a daily journal can have different types of bullets:\n",
    "  o event\n",
    "  . task\n",
    "  x completed task\n",
    "  - note\n",
    "  < moved to future log\n",
    "  > moved to another daily, weekly, or monthly note\n",
    "*   important (can be added left of another bullet)\n",
    "\n",
    "Here are examples of what a daily note looks like:\n",
    "\n",
    "1030 SUN\n",
    "  o Halloween Party @ 7PM @ Kid's school \n",
    "  . Pick up candy\n",
    "  x Buy pumpkins\n",
    "* > Call mom\n",
    "\n",
    "1031 MON\n",
    "  - Trickortreaters were so cute\n",
    "  . Call mom\n",
    "\n",
    "\n",
    "Your task is to rewrite the journal in a consistent format and fix mistakes in\n",
    "the OCR. Only answer with the corrected text, and no other commentary, as the\n",
    "answer will be programtically parsed\n",
    "\"\"\"\n",
    "\n",
    "response = model.generate_content([IMG_PROMPT, page_png])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0201 THU\n",
      "x act rent\n",
      "< ai through put notes\n",
      "- Board overview/videos?\n",
      "*x Ask Photos user study\n",
      "**x run Synthetic data colab \n",
      "**x notes for Synthetic data\n",
      "**x review TT Databases slides\n",
      "*x Wrap up DMA rollout plan\n",
      "x Summarize synth data next\n",
      "< monthly reflection\n",
      "* < query lens transitions\n",
      "\n",
      "0202 FRI\n",
      "**< Serving TPU calendar\n",
      "*x Tristan email intro \n",
      "*x 15 mins; DMA rollout plan Cleanup\n",
      "*x synth data next steps\n",
      "x gap print return label\n",
      "*x AI through put bullet points\n",
      "*x review quality report\n",
      "\n",
      "0203 SAT 0204 SUN\n",
      "- Booked flights to LOW\n",
      "*x TP stand\n",
      "* > Fill cavity \n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(response.text)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
