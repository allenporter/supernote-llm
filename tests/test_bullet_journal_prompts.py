"""Test that the bullet journal prompts are parsed correctly."""

from dataclasses import asdict
import pathlib
import yaml

from syrupy import SnapshotAssertion
import pytest

from bullet_journal_llm.model import Prompt, JournalPage, DynamicPrompt

PROMPTS_DIR = pathlib.Path("bullet_journal_llm/dynamic_prompts")
PROMPT_FILES = list(PROMPTS_DIR.glob("*.yaml"))


@pytest.mark.parametrize("filename", PROMPT_FILES, ids=[f.name for f in PROMPT_FILES])
def test_parse_dynamic_prompts(filename: pathlib.Path, snapshot: SnapshotAssertion) -> None:
    """Test that the bullet journal prompts are parsed correctly."""

    dynamic_prompt = DynamicPrompt.from_file(filename)
    assert dynamic_prompt.prompt.to_dict() == snapshot

    for page in dynamic_prompt.pages:
        assert page.to_dict() == snapshot
