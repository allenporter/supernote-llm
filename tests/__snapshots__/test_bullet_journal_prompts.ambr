# serializer version: 1
# name: test_parse_dynamic_prompts[daily.yaml]
  '''
  content: "1030 SUN\n  o Halloween Party @ 7PM @ Kid's school \n  \u2022 Pick up candy\n
    \ X Buy pumpkins\n* > Call mom\n\n1031 MON\n  - Trickortreaters were so cute\n*
    \u2022 Call mom\n"
  created_at: '2022-10-30T21:07:60.068713'
  filename: Daily-00-P20221030210760068713clbdtpKcEWTi.png
  prompt: Below is an example of some OCR text for a daily log for October 30th and
    October 31st, 2022. The output json has a separate rapid log record for the date
    it was logged under. Note that this is just one example and your task will be to
    do this from an actual png image.
  
  '''
# ---
# name: test_parse_dynamic_prompts[daily.yaml].1
  '''
  created_at: '2022-10-30T21:07:60.068713'
  date: '2022-10-30'
  filename: Daily-01-P20221030210760068713clbdtpKcEWTi.png
  label: daily
  records:
  - content: Halloween Party @ 7PM @ Kid's school
    date: '2022-10-30'
    status: migrated
    type: event
  - content: Pick up candy
    date: '2022-10-30'
    type: note
  - content: Buy pumpkins
    date: '2022-10-30'
    status: completed
    type: task
  - content: Call mom
    critical: true
    date: '2022-10-30'
    status: migrated
    type: task
  - content: Trickortreaters were so cute
    date: '2022-10-31'
    type: note
  - content: Call mom
    critical: true
    date: '2022-10-31'
    status: migrated
    type: task
  
  '''
# ---
# name: test_parse_dynamic_prompts[default.yaml]
  '''
  prompt: "You are analyzing PNG images of handwritten text from an\ne-ink notebook
    SuperNote A5X. The notes are written in English and are in a\nbullet journal format.
    You can see that the text is not perfect and will need\nsome cleaning up\n\nThe
    Bullet Journal method is a system that combines elements of mindfulness,\nproductivity,
    and self-discovery. It empowers the user to become the author of their\nown life,
    allowing them to track the past, organize the present, and plan for the\nfuture.
    A Bullet Journal method may be described as a productivity system or an\norganization
    system, but at its core, the method is a tool for changing the way\nwe approach
    our day-to-day tasks and long term goals. The Bullet Journal method\nis centered
    on one key idea: intentionality. Why do we do what we do? What makes\nthese goals
    meaningful to us? What tasks are the most relevant to  us at any\ngiven point in
    time? \n\nRapid logging is the language of the bullet journal method and it functions\nthrough
    the use of  Bullets to indicate a task's status. A task starts with a simple\ndot
    \"\u2022\" to represent a task. If a task is completed, mark it with an \"X\". If
    it's\nmigrated to a future date, use a right arrow (>) to indicate that. And additional\nbullet
    styles can be used depending on what makes sense to the author.\n\nTasks within
    the Bullet Journal method can then fall within any of the logs used\ndepending on
    where they fall in the author's timeline. Typically, journals contain\na Daily Log,
    Weekly Log, and Monthly Log.\n\nDaily Log\n---------\nThe daily log is used to record
    tasks, events, and notes on a day-to-day\nbasis. A daily log is simply the current
    date as the header and then a list of\nlog entires. They can be short, bulleted
    entries with symbols to represent\ndifferent types of content (tasks, events, notes).\n\nWhen
    new tasks arise throughout the day, they are add them to the daily log. If\na task
    is completed, it is marked with an \"X\". If a task is not completed and the\nauthor
    still wants to work on it, they can migrate it to a future date by using\na right
    arrow (>) to indicate that. This way, the author maintains a concise and up-to-date\nrecord
    of their daily activities.\n\nWeekly Log\n----------\nThe weekly log provides an
    overview of the week, allowing the author to plan and\norganize tasks, events, and
    commitments in a broader context. A weekly log \nis started by writing the dates
    of the week at the top of the page. Below are\ntypically a list of tasks and notes
    for the week, using bulleted entries. At\nthe end of the week, the author can migrate
    unfinished tasks to the next week\nas needed.\n\nThe weekly log serves as a snapshot
    of the upcoming week, helping prioritize\ntasks, allocate time, and have a holistic
    view of the author's commitments.\n\nA weekly log may also be accompanied by a weekly
    review, where the author\nreflects on the previous week's accomplishments, challenges,
    and goals for the upcoming\nweek.\n\nMonthly Log\n-----------\nThe monthly log offers
    a broader perspective, enabling the author to plan and\ntrack their activities,
    goals, and events throughout the month. To create a monthly\nlog, designate a double-page
    spread in their notebook for each month.\n\nA monthly spread has the name of the
    month at the top of the list. The left-hand\npage has a list of dates (1-30/31)
    down the side. This space is used to mark important\nevents, deadlines, or appointments
    for each day. The right-hand page is used as\na monthly last list, where the author
    can list out tasks, goals, or projects they\nwant to focus on during the month.\n\nThe
    monthly log serves as a reference point, allowing the author to see the big\npicture
    and plan ahead. As the month progresses, they can refer to this log, migrate\ntasks
    to specific days in their daily log or add new tasks as they arise.\n\nFuture Log\n----------\nA
    Future Log is designed to help the author plan and organize events, tasks,\nand
    important dates that extend beyond your current month. It serves as a\nforward-looking
    overview of the upcoming months, allowing the author to capture\nand track future
    commitments in a structured manner. The Future Log is a way\nto ensure important
    events or deadlines are not forgotten and time and resources\ncan be allocated effectively.
    The format is usually a month name and a list of\ntasks or events scheduled for
    that month.\n"
  
  '''
# ---
# name: test_parse_dynamic_prompts[monthly-01.yaml]
  '''
  content: "FEBRUARY REVIEW\n\n- What was the most memorable part of last month?\n  -
    The trip to the mountains with the family\n  - The team-building exercise at work\n\n-
    What were the three biggest lessons learned?\n  - The importance of taking time
    off & quality time with family\n  - The need to prioritize self-care\n  - The value
    of open communication\n\n- What are my goals for the upcoming month?\n  - Plan another
    weekend getaway with the family\n  - Spend at least one day a week outside in nature\n
    \ - Schedule a meeting with my manager to discuss career goals\n"
  created_at: '2023-03-01T06:30:52.75344'
  filename: Monthly-02-P2023030106305275344.png
  prompt: There is a special "Monthly Review" page. This is used for reflection on the
    previous monthy as well as the month ahead, usually in the form of prompt questions.
    Typicaly the monthly reflection is a time to check in on progress on quarterly or
    yearly goals, and to set new goals for the upcoming month.
  
  '''
# ---
# name: test_parse_dynamic_prompts[monthly-01.yaml].1
  '''
  created_at: '2023-03-01T06:30:52.75344'
  date: '2023-03-01'
  filename: Monthly-02-P2023030106305275344.png
  label: monthly
  records:
  - content: What was the most memorable part of last month?
    entries:
    - The trip to the mountains with the family
    - The team-building exercise at work
    type: note
  - content: What were the three biggest lessons learned?
    entries:
    - The importance of taking time off & quality time with family
    - The need to prioritize self-care
    - The value of open communication
    type: note
  - content: What are my goals for the upcoming month?
    entries:
    - Plan another weekend getaway with the family
    - Spend at least one day a week outside in nature
    - Schedule a meeting with my manager to discuss career goals
    type: note
  
  '''
# ---
# name: test_parse_dynamic_prompts[monthly-02.yaml]
  '''
  content: 'MARCH 2023
  
  
    1 W
  
    2 T
  
    3 F
  
    4 S
  
    5 S
  
    6 M
  
    7 T
  
    8 W o Infrastructure Virtual Summit
  
    9 T
  
    10 F
  
    11 S
  
    12 S
  
    13 M
  
    14 T
  
    15 W
  
    16 T
  
    17 F o St. Patrick''s Day
  
    18 S
  
    19 S
  
    20 M o Work Trip NYC
  
    21 T o Work Trip NYC
  
    22 W o Work Trip NYC o Return flight to SF
  
    23 T
  
    24 F
  
    25 S
  
    26 S
  
    27 M o Kids no School
  
    28 T o Kids no School
  
    29 W o Kids no School
  
    30 T o Kids no School
  
    31 F o Kids no School
  
    '
  created_at: '2023-03-01T06:42:60.113420'
  filename: Monthly-02-P20230301064260113420.png
  prompt: The is an example of the calendar part of a Monthly Log which is a list of
    dates and weekday names where import event, tasks, or notes can be recorded to get
    a sense of the month at a glance.
  
  '''
# ---
# name: test_parse_dynamic_prompts[monthly-02.yaml].1
  '''
  created_at: '2023-03-01T06:42:60.113420'
  date: '2023-03-01'
  filename: Monthly-02-P20230301064260113420.png
  label: monthly
  records:
  - content: Infrastructure Virtual Summit
    date: '2023-03-08'
    type: event
  - content: St. Patrick's Day
    date: '2023-03-17'
    type: event
  - content: Work Trip NYC
    date: '2023-03-20'
    type: event
  - content: Work Trip NYC
    date: '2023-03-21'
    type: event
  - content: Work Trip NYC
    date: '2023-03-22'
    type: event
  - content: Return flight to SF
    date: '2023-03-22'
    type: event
  - content: Kids no School
    date: '2023-03-27'
    type: event
  - content: Kids no School
    date: '2023-03-28'
    type: event
  - content: Kids no School
    date: '2023-03-29'
    type: event
  - content: Kids no School
    date: '2023-03-30'
    type: event
  - content: Kids no School
    date: '2023-03-31'
    type: event
  
  '''
# ---
# name: test_parse_dynamic_prompts[monthly-03.yaml]
  '''
  content: "MONTHLY 2023-03\n\nX Plan weekend getaway with family\n\u2022 Look up hiking
    trails in the area\n< Schedule meeting with manager to discuss career goals\n\u2022
    Squeaky garage door\n*X Call mom\nX Finish salary report\n\u2022 Finish reading
    \"Lessons in Chemistry\"\n"
  created_at: '2023-03-01T06:42:60.113420'
  filename: Monthly-03-P20230301064260113420.png
  prompt: The monthly log will be a snapshot of tasks to accomplish during the month.
    This is in the normal rapid logging format with tasks.
  
  '''
# ---
# name: test_parse_dynamic_prompts[monthly-03.yaml].1
  '''
  created_at: '2023-03-01T06:42:60.113420'
  date: '2023-03-01'
  filename: Monthly-03-P20230301064260113420.png
  label: monthly
  records:
  - content: Plan weekend getaway with family
    status: completed
    type: task
  - content: Look up hiking trails in the area
    type: task
  - content: Schedule meeting with manager to discuss career goals
    type: task
  - content: Squeaky garage door
    type: task
  - content: Call mom
    critical: true
    status: completed
    type: task
  - content: Finish salary report
    status: completed
    type: task
  - content: Finish reading "Lessons in Chemistry"
    type: task
  
  '''
# ---
# name: test_parse_dynamic_prompts[profile.yaml]
  '''
  prompt: 'Author: Allen Porter, software engineer from Los Angeles, CA. Uses the notebook
    to track work, hobbies, and family obligations.'
  
  '''
# ---
# name: test_parse_dynamic_prompts[rapid_log_legend.yaml]
  '''
  prompt: "Here is a legend for the different types of tasks in a rapid log:\n  o event\n
    \ \u2022 task\n* \u2022 critical task\n  X completed task\n  - note\n  < migrated
    to the future log\n  > migrated the task forward to another daily, weekly, or monthly
    log\n\nNote that * can be added to any task to indicate it is a critical task (e.g.
    a\ncritical task can be completed or migrated)\n"
  
  '''
# ---
# name: test_parse_dynamic_prompts[weekly-01.yaml]
  '''
  content: "WEEKLY REVIEW 0826-0901\n\n- What did I accomplish last week?\n  - After
    completing the salary analysis, i got assigned to do the final report\n  - I also
    got a lot of positive feedback from the team on the presentation\n- What were my
    biggest challenges?\n  - I had a hard time getting the data for the salary report\n
    \ - Perhaps I should consider asking for help\n- What are my goals for next week?\n
    \ - Make sure i talk to at least one parent\n  - Finish the salary report\n"
  created_at: '2024-09-02T07:15:10.7646'
  filename: Weeky-07-P2024090207100618749.png
  prompt: 'There is a special "Weekly Review" page. This is used for reflection on the
    previous week as well as the week ahead, and is usually in the form of prompt questions. '
  
  '''
# ---
# name: test_parse_dynamic_prompts[weekly-01.yaml].1
  '''
  created_at: '2024-09-02T07:15:10.7646'
  date: '2024-09-01'
  filename: Weeky-08-P2024090207150618749.png
  label: weekly
  records:
  - content: What did I accomplish last week?
    entries:
    - After completing the salary analysis, i got assigned to do the final report
    - I also got a lot of positive feedback from the team on the presentation
    type: note
  - content: What were my biggest challenges?
    entries:
    - I had a hard time getting the data for the salary report
    - Perhaps I should consider asking for help
    type: note
  - content: What are my goals for next week?
    entries:
    - Make sure i talk to at least one parent
    - Finish the salary report
    type: note
  
  '''
# ---
# name: test_parse_dynamic_prompts[weekly-02.yaml]
  '''
  content: "WEEKLY 0902-0908\n\nPersonal\n  X Buy groceries\n  \u2022 Go to the gym
    3x times this week X X \u2022\n  \u2022 Call mom\n  < Clean out closet\n\nWork\n*
    X Initial draft of salary report\n  \u2022 Check if Kendra is available to meet
    this week\n  < Send out report by Wednesday\n"
  created_at: '2024-09-02T07:15:10.7646'
  filename: Weeky-08-P2024090207150618749.png
  prompt: The weekly log will be a snapshot of tasks to accomplish during the week.
    This is in the normal rapid logging format with tasks and they may be grouped into
    different categories of tasks.
  
  '''
# ---
# name: test_parse_dynamic_prompts[weekly-02.yaml].1
  '''
  created_at: '2024-09-02T07:15:10.7646'
  date: '2024-09-02'
  filename: Weeky-08-P202409020715107646.png
  label: weekly
  records:
  - content: Buy groceries
    label: personal
    status: completed
    type: note
  - content: Go to the gym 3x
    label: personal
    status: completed
    type: task
  - content: Call mom
    label: personal
    type: task
  - content: Clean out closet
    label: personal
    status: migrated
    type: task
  - content: Initial draft of salary report
    label: work
    status: completed
    type: task
  - content: Check if Kendra is available to meet this week
    label: work
    type: task
  - content: Send out report by Wednesday
    label: work
    status: migrated
    type: task
  
  '''
# ---
